sigstore-java/src/test/java/dev/sigstore/KeylessSignerTest.java

```diff
@@ -127,20 +127,19 @@ public void sign_failGithubOidcCheck() throws Exception {
   @EnabledIfOidcExists(provider = OidcProviderType.GITHUB)
   // this test will only pass on the github.com/sigstore/sigstore-java repository
   public void sign_passGithubOidcCheck() throws Exception {
-    // silly way to get the right oidc identity to make sure our simple matcher works
-    var jws =
-        JsonWebSignature.parse(
-            new GsonFactory(),
-            GithubActionsOidcClient.builder().build().getIDToken(System.getenv()).getIdToken());
-    var expectedGithubSubject = jws.getPayload().getSubject();
     var signer =
         KeylessSigner.builder()
             .sigstorePublicDefaults()
             .allowedOidcIdentities(
                 List.of(
                     OidcIdentity.of(
-                        expectedGithubSubject, "https://token.actions.githubusercontent.com"),
-                    OidcIdentity.of("some@other.com", "https://accounts.other.com")))
+                        StringMatchable.of(
+                            Pattern.compile(
+                                "https://github\\.com/sigstore/sigstore-java/\\.github/workflows/.*\\.yaml@.*")),
+                        StringMatchable.of("https://token.actions.githubusercontent.com")),
+                    OidcIdentity.of(
+                        StringMatchable.of("some@other.com"),
+                        StringMatchable.of("https://accounts.other.com"))))
```

Comment:
Have you considered `allowedOidcIdentities(List<? extends Predicate<? super OidcIdentity>>)` instead?